--- gst-plugins-bad-1.16.0/configure.ac.orig	2019-05-20 12:19:22.139787834 +0200
+++ gst-plugins-bad-1.16.0/configure.ac	2019-05-20 12:23:00.638290319 +0200
@@ -1067,8 +1067,8 @@
             [AS_IF([test "x$MFX_HOME" != "x"],
                    [MSDK_PREFIX="$MFX_HOME"],
                    [MSDK_PREFIX="/opt/intel/media"])])
-    MSDK_CFLAGS="-I$MSDK_PREFIX/include -I$MSDK_PREFIX/include/mfx"
-    MSDK_LIBS="-L$MSDK_PREFIX/lib/lin_x64 -L$MSDK_PREFIX/lib/x64 -L$MSDK_PREFIX/lib64 -L$MSDK_PREFIX/lib -lmfx -ldl"
+    MSDK_CFLAGS="-I/usr/include/mfx"
+    MSDK_LIBS="-lmfx -ldl"
     AC_SUBST(MSDK_CFLAGS)
     AC_SUBST(MSDK_LIBS)
 
