--- gst-plugins-bad-1.26.0/sys/aja/meson.build.orig	2025-03-11 21:14:44.000000000 +0100
+++ gst-plugins-bad-1.26.0/sys/aja/meson.build	2025-03-24 18:46:58.446763248 +0100
@@ -77,7 +77,6 @@ foreach dir : aja_include_dirs
                                      f'@dir@',
                                      f'@dir@/ajaanc/includes',
                                      f'@dir@/ajantv2/includes',
-                                     f'@dir@/ajantv2/src/lin',
                                     ],
                                     is_system : true)
 
