--- gst-plugins-bad-0.10.5/configure.ac.orig	2007-06-24 13:21:54.353070000 +0200
+++ gst-plugins-bad-0.10.5/configure.ac	2007-06-24 19:54:50.024571623 +0200
@@ -679,7 +679,7 @@
 dnl *** MythTV ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_MYTHTV, true)
 AG_GST_CHECK_FEATURE(MYTHTV, [MythTV client plugins], mythtvsrc, [
-  PKG_CHECK_MODULES(GMYTH, gmyth-0.1 >= 0.1.0.3, HAVE_MYTHTV="yes", [
+  PKG_CHECK_MODULES(GMYTH, gmyth >= 0.1.0.3, HAVE_MYTHTV="yes", [
     HAVE_MYTHTV="no"
     AC_MSG_RESULT(no)
   ])
--- gst-plugins-bad-0.10.5/ext/mythtv/gstmythtvsrc.c.orig	2007-05-31 21:12:37.000000000 +0200
+++ gst-plugins-bad-0.10.5/ext/mythtv/gstmythtvsrc.c	2007-06-24 20:15:26.903057312 +0200
@@ -345,10 +345,10 @@
         read = -1;
         goto eos;
       } else {
-        if (len == GMYTHTV_FILE_TRANSFER_READ_ERROR) {  /* -314 */
+        if (len == GMYTH_FILE_READ_ERROR) {  /* -314 */
           GST_INFO_OBJECT (src, "[LiveTV] FileTransfer READ_ERROR!");
           goto done;
-        } else if (len == GMYTHTV_FILE_TRANSFER_NEXT_PROG_CHAIN) {      /* -315 */
+        } else if (len == GMYTH_FILE_READ_NEXT_PROG_CHAIN) {      /* -315 */
           GST_INFO_OBJECT (src,
               "[LiveTV] FileTransfer - Go to the next program chain!");
           continue;
@@ -650,28 +650,27 @@
   if (src->live_tv || gmyth_uri_is_livetv (gmyth_uri)) {
     gint ch;
 
-    src->spawn_livetv = gmyth_livetv_new ();
+    src->spawn_livetv = gmyth_livetv_new (src->backend_info);
 
     ch = gmyth_uri_get_channel_num (gmyth_uri);
     if (ch != -1)
       src->channel_num = ch;
 
     if (src->channel_num != GST_GMYTHTV_CHANNEL_DEFAULT_NUM) {
-      if (gmyth_livetv_channel_setup (src->spawn_livetv, src->channel_num,
-              src->backend_info) == FALSE) {
+      if (gmyth_livetv_channel_setup (src->spawn_livetv, src->channel_num) == FALSE) {
         GST_INFO_OBJECT (src, "LiveTV setup felt down on error");
         ret = FALSE;
         goto init_failed;
       }
     } else {
-      if (gmyth_livetv_setup (src->spawn_livetv, src->backend_info) == FALSE) {
+      if (gmyth_livetv_setup (src->spawn_livetv) == FALSE) {
         GST_INFO_OBJECT (src, "LiveTV setup felt down on error");
         ret = FALSE;
         goto init_failed;
       }
     }
 
-    src->file_transfer = gmyth_livetv_create_file_transfer (src->spawn_livetv);
+    src->file_transfer = (GMythFileTransfer*) gmyth_livetv_create_file_transfer (src->spawn_livetv);
 
     if (NULL == src->file_transfer) {
       GST_INFO_OBJECT (src, "[LiveTV] FileTransfer equals to NULL");
@@ -703,9 +703,9 @@
 
   GST_INFO_OBJECT (src,
       "MythTV FileTransfer filesize = %lld, content_size = %lld!",
-      src->file_transfer->filesize, src->content_size);
+      gmyth_file_transfer_get_filesize(src->file_transfer), src->content_size);
 
-  src->content_size = src->file_transfer->filesize;
+  src->content_size = gmyth_file_transfer_get_filesize(src->file_transfer);
 
   src->do_start = FALSE;
 
@@ -839,7 +839,7 @@
   }
   src->content_size_last = src->content_size;
 
-  src->content_size = src->file_transfer->filesize;
+  src->content_size = gmyth_file_transfer_get_filesize(src->file_transfer);
   if (src->live_tv) {
     src->wait_to_transfer = 0;
     while (src->wait_to_transfer++ < GMYTHTV_TRANSFER_MAX_WAITS &&
