--- gst-plugins-bad-0.10.3/configure.ac.orig	2006-05-21 10:37:39.070248000 +0200
+++ gst-plugins-bad-0.10.3/configure.ac	2006-05-21 20:18:41.220996750 +0200
@@ -777,7 +777,7 @@
 dnl *** soundtouch ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_SOUNDTOUCH, true)
 AG_GST_CHECK_FEATURE(SOUNDTOUCH, [soundtouch plug-in], soundtouch, [
-  PKG_CHECK_MODULES(SOUNDTOUCH, libSoundTouch, HAVE_SOUNDTOUCH=yes, [
+  PKG_CHECK_MODULES(SOUNDTOUCH, soundtouch-1.0, HAVE_SOUNDTOUCH=yes, [
     HAVE_SOUNDTOUCH=no
      AC_MSG_RESULT(no)])
   AC_SUBST(SOUNDTOUCH_CFLAGS)

