--- gst-plugins-bad-0.10.21/ext/celt/gstceltenc.c~	2010-12-07 17:22:30.000000000 +0100
+++ gst-plugins-bad-0.10.21/ext/celt/gstceltenc.c	2011-02-27 20:27:44.205703990 +0100
@@ -629,7 +629,9 @@
   if (!enc->mode)
     goto mode_initialization_failed;
 
-#ifdef HAVE_CELT_0_7
+#ifdef HAVE_CELT_0_8
+  celt_header_init (&enc->header, enc->mode, enc->frame_size, enc->channels);
+#elif defined(HAVE_CELT_0_7)
   celt_header_init (&enc->header, enc->mode, enc->channels);
 #else
   celt_header_init (&enc->header, enc->mode);
