diff -aurN gst-plugins-bad-0.10.3.orig/ext/neon/gstneonhttpsrc.c gst-plugins-bad-0.10.3/ext/neon/gstneonhttpsrc.c
--- gst-plugins-bad-0.10.3.orig/ext/neon/gstneonhttpsrc.c	2006-05-02 18:16:32.000000000 +0200
+++ gst-plugins-bad-0.10.3/ext/neon/gstneonhttpsrc.c	2006-06-22 19:36:02.614479536 +0200
@@ -627,7 +627,7 @@
     goto clear;
   }
 
-  if (!parsed->path || parsed->authinfo) {
+  if (!parsed->path || parsed->userinfo) {
     goto clear;
   }
 
