--- gst-plugins-bad-1.0.2/docs/plugins/gst-plugins-bad-plugins-docs.sgml.orig	2012-10-24 14:43:23.000000000 +0200
+++ gst-plugins-bad-1.0.2/docs/plugins/gst-plugins-bad-plugins-docs.sgml	2012-10-28 12:53:08.981375217 +0100
@@ -116,7 +116,6 @@
     <xi:include href="xml/plugin-dtsdec.xml" />
     <xi:include href="xml/plugin-dvb.xml" />
     <xi:include href="xml/plugin-dvdspu.xml" />
-    <xi:include href="xml/plugin-eglglessink.xml" />
     <xi:include href="xml/plugin-faac.xml" />
     <xi:include href="xml/plugin-faad.xml" />
     <xi:include href="xml/plugin-festival.xml" />
