--- gst-plugins-bad-0.10.9/ext/celt/gstceltenc.c.orig	2008-12-11 21:40:48.956289465 +0100
+++ gst-plugins-bad-0.10.9/ext/celt/gstceltenc.c	2008-12-11 21:40:21.072297008 +0100
@@ -741,7 +741,7 @@
     GST_DEBUG_OBJECT (enc, "encoding %d samples (%d bytes)", frame_size, bytes);
 
     outsize =
-        celt_encode (enc->state, data, GST_BUFFER_DATA (outbuf),
+        celt_encode (enc->state, data, NULL, GST_BUFFER_DATA (outbuf),
         bytes_per_packet);
 
     g_free (data);
