--- gst-plugins-bad-0.10.4/configure.ac.orig	2006-12-25 02:57:43.050298000 +0100
+++ gst-plugins-bad-0.10.4/configure.ac	2006-12-25 03:07:24.491432656 +0100
@@ -261,7 +261,7 @@
     AC_MSG_NOTICE([cannot find X11 development files])
   else
     dnl this is much more than we want
-    X_LIBS="$X_LIBS $X_PRE_LIBS $X_EXTRA_LIBS"
+    X_LIBS="$X_LIBS $X_EXTRA_LIBS"
     dnl AC_PATH_XTRA only defines the path needed to find the X libs,
     dnl it does not add the libs; therefore we add them here
     X_LIBS="$X_LIBS -lX11"
--- gst-plugins-bad-0.10.4/ext/cdaudio/Makefile.am.orig	2004-10-27 02:21:33.000000000 +0200
+++ gst-plugins-bad-0.10.4/ext/cdaudio/Makefile.am	2006-12-25 03:48:22.479505414 +0100
@@ -2,5 +2,5 @@
 
 libgstcdaudio_la_SOURCES = gstcdaudio.c 
 libgstcdaudio_la_CFLAGS = $(GST_CFLAGS)
-libgstcdaudio_la_LIBADD =
+libgstcdaudio_la_LIBADD = $(GST_LIBS)
 libgstcdaudio_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS) -lcdaudio
