--- gst-plugins-bad-0.10.6/configure.ac~	2008-02-21 16:58:30.000000000 +0100
+++ gst-plugins-bad-0.10.6/configure.ac	2008-02-21 17:09:37.048926286 +0100
@@ -733,13 +733,13 @@
 translit(dnm, m, l) AM_CONDITIONAL(USE_NAS, true)
 AG_GST_CHECK_FEATURE(NAS, [nas plug-in], nassink, [
   HAVE_NAS="no"
-  if test "x$HAVE_X" = "xyes"; then
+#  if test "x$HAVE_X" = "xyes"; then
     save_cppflags=$CFLAGS
     CPPFLAGS="$CPPFLAGS $X_CFLAGS"
     AG_GST_CHECK_LIBHEADER(NAS, audio, AuOpenServer, $X_LIBS, audio/audiolib.h,
       NAS_LIBS="$X_LIBS -laudio" NAS_CFLAGS="$X_CFLAGS")
     CPPFLAGS="$save_cppflags"
-  fi
+#  fi
   AC_SUBST(NAS_CFLAGS)
   AC_SUBST(NAS_LIBS)
 ])
