--- gst-plugins-bad-1.24.0/sys/aja/gstajasrc.cpp.orig	2024-03-05 00:51:42.000000000 +0100
+++ gst-plugins-bad-1.24.0/sys/aja/gstajasrc.cpp	2024-03-21 06:03:23.822345459 +0100
@@ -1058,9 +1058,9 @@ static gboolean gst_aja_src_configure(Gs
       // their framebuffers/muxers, and muxers from their framebuffers
       for (auto iter = connections.begin(); iter != connections.end(); iter++) {
         if (iter->first == NTV2_XptFrameBuffer1Input ||
-            iter->first == NTV2_XptFrameBuffer1BInput ||
+            iter->first == NTV2_XptFrameBuffer1DS2Input ||
             iter->first == NTV2_XptFrameBuffer2Input ||
-            iter->first == NTV2_XptFrameBuffer2BInput ||
+            iter->first == NTV2_XptFrameBuffer2DS2Input ||
             iter->second == NTV2_Xpt425Mux1AYUV ||
             iter->second == NTV2_Xpt425Mux1BYUV ||
             iter->second == NTV2_Xpt425Mux2AYUV ||
@@ -1078,10 +1078,8 @@ static gboolean gst_aja_src_configure(Gs
     } else if (self->channel == NTV2_CHANNEL1) {
       for (auto iter = connections.begin(); iter != connections.end(); iter++) {
         if (iter->first == NTV2_XptFrameBuffer1Input ||
-            iter->first == NTV2_XptFrameBuffer1BInput ||
             iter->first == NTV2_XptFrameBuffer1DS2Input ||
             iter->first == NTV2_XptFrameBuffer2Input ||
-            iter->first == NTV2_XptFrameBuffer2BInput ||
             iter->first == NTV2_XptFrameBuffer2DS2Input ||
             iter->second == NTV2_Xpt425Mux1AYUV ||
             iter->second == NTV2_Xpt425Mux1BYUV ||
@@ -1104,10 +1102,8 @@ static gboolean gst_aja_src_configure(Gs
     } else if (self->channel == NTV2_CHANNEL5) {
       for (auto iter = connections.begin(); iter != connections.end(); iter++) {
         if (iter->first == NTV2_XptFrameBuffer5Input ||
-            iter->first == NTV2_XptFrameBuffer5BInput ||
             iter->first == NTV2_XptFrameBuffer5DS2Input ||
             iter->first == NTV2_XptFrameBuffer6Input ||
-            iter->first == NTV2_XptFrameBuffer6BInput ||
             iter->first == NTV2_XptFrameBuffer6DS2Input ||
             iter->second == NTV2_Xpt425Mux3AYUV ||
             iter->second == NTV2_Xpt425Mux3BYUV ||
@@ -1208,9 +1204,9 @@ static gboolean gst_aja_src_configure(Gs
   if (self->quad_mode) {
     if (self->input_source >= GST_AJA_INPUT_SOURCE_HDMI1 &&
         self->input_source <= GST_AJA_INPUT_SOURCE_HDMI4) {
-      router.AddConnection(NTV2_XptFrameBuffer1BInput, NTV2_Xpt425Mux1BYUV);
+      router.AddConnection(NTV2_XptFrameBuffer1DS2Input, NTV2_Xpt425Mux1BYUV);
       router.AddConnection(NTV2_XptFrameBuffer2Input, NTV2_Xpt425Mux2AYUV);
-      router.AddConnection(NTV2_XptFrameBuffer2BInput, NTV2_Xpt425Mux2BYUV);
+      router.AddConnection(NTV2_XptFrameBuffer2DS2Input, NTV2_Xpt425Mux2BYUV);
 
       router.AddConnection(NTV2_Xpt425Mux1AInput, NTV2_XptHDMIIn1);
       router.AddConnection(NTV2_Xpt425Mux1BInput, NTV2_XptHDMIIn1Q2);
@@ -1250,11 +1246,11 @@ static gboolean gst_aja_src_configure(Gs
           // TSI?
         } else {
           if (self->channel == NTV2_CHANNEL1) {
-            router.AddConnection(NTV2_XptFrameBuffer1BInput,
+            router.AddConnection(NTV2_XptFrameBuffer1DS2Input,
                                  NTV2_Xpt425Mux1BYUV);
             router.AddConnection(NTV2_XptFrameBuffer2Input,
                                  NTV2_Xpt425Mux2AYUV);
-            router.AddConnection(NTV2_XptFrameBuffer2BInput,
+            router.AddConnection(NTV2_XptFrameBuffer2DS2Input,
                                  NTV2_Xpt425Mux2BYUV);
 
             router.AddConnection(NTV2_Xpt425Mux1AInput, NTV2_XptSDIIn1);
@@ -1262,11 +1258,11 @@ static gboolean gst_aja_src_configure(Gs
             router.AddConnection(NTV2_Xpt425Mux2AInput, NTV2_XptSDIIn3);
             router.AddConnection(NTV2_Xpt425Mux2BInput, NTV2_XptSDIIn4);
           } else if (self->channel == NTV2_CHANNEL5) {
-            router.AddConnection(NTV2_XptFrameBuffer5BInput,
+            router.AddConnection(NTV2_XptFrameBuffer5DS2Input,
                                  NTV2_Xpt425Mux3BYUV);
             router.AddConnection(NTV2_XptFrameBuffer6Input,
                                  NTV2_Xpt425Mux4AYUV);
-            router.AddConnection(NTV2_XptFrameBuffer6BInput,
+            router.AddConnection(NTV2_XptFrameBuffer6DS2Input,
                                  NTV2_Xpt425Mux4BYUV);
 
             router.AddConnection(NTV2_Xpt425Mux3AInput, NTV2_XptSDIIn5);
